### PR TITLE
install all operators within one script

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: hack/common.sh [hunk 1]
```diff
@@ -89,6 +89,7 @@ METRICS_EXPORTER_FULL_IMAGE_NAME="${METRICS_EXPORTER_FULL_IMAGE_NAME:-${DEFAULT_
 UX_BACKEND_OAUTH_FULL_IMAGE_NAME="${UX_BACKEND_OAUTH_FULL_IMAGE_NAME:-${DEFAULT_UX_BACKEND_OAUTH_FULL_IMAGE_NAME}}"
 
 CSI_ADDONS_CATALOG_FULL_IMAGE_NAME="quay.io/ocs-dev/csi-addons-catalog:v0.8.0"
+CSI_ADDONS_BUNDLE_FULL_IMAGE_NAME="quay.io/csiaddons/csi-addons-catalog:v0.8.0"
```
Comment (malayparida2000) [inline hack/common.sh:92]: This image doesn't exist

Comment (iamniting) [reply hack/common.sh]: my bad changed the image.

**Comment by malayparida2000** [comment]
I tested the changes on my cluster & it looks good now. Here are the output
```
$ make install                                           
Ensuring operator-sdk
hack/ensure-operator-sdk.sh
Using operator-sdk CLI present at /Users/mparida/go/src/github.com/red-hat-storage/ocs-operator/bin/operator-sdk-v1.25.4
Installing operators
hack/install-ocs-operator.sh
Namespace "openshift-storage" does not exist: creating it
namespace/openshift-storage created
configmap/ocs-operator-config created
configmap/ocs-client-operator-config created
INFO[0020] Creating a File-Based Catalog of the bundle "quay.io/ocs-dev/rook-ceph-operator-bundle:master-75180f4f3" 
INFO[0023] Generated a valid File-Based Catalog         
INFO[0029] Created registry pod: quay-io-ocs-dev-rook-ceph-operator-bundle-master-75180f4f3 
INFO[0029] Created CatalogSource: rook-ceph-operator-catalog 
INFO[0030] OperatorGroup "operator-sdk-og" created      
INFO[0030] Created Subscription: rook-ceph-operator-v4-17-0-sub 
INFO[0041] Approved InstallPlan install-ncnsm for the Subscription: rook-ceph-operator-v4-17-0-sub 
INFO[0041] Waiting for ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" to reach 'Succeeded' phase 
INFO[0042]   Waiting for ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" to appear 
INFO[0043]   Found ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" phase: Pending 
INFO[0058]   Found ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" phase: InstallReady 
INFO[0059]   Found ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" phase: Installing 
INFO[0061]   Found ClusterServiceVersion "openshift-storage/rook-ceph-operator.v4.17.0" phase: Succeeded 
INFO[0061] OLM has successfully installed "rook-ceph-operator.v4.17.0" 
INFO[0032] Creating a File-Based Catalog of the bundle "quay.io/csiaddons/k8s-bundle:v0.8.0" 
INFO[0037] Generated a valid File-Based Catalog         
INFO[0040] Created registry pod: quay-io-csiaddons-k8s-bundle-v0-8-0 
INFO[0040] Created CatalogSource: csi-addons-catalog    
INFO[0041] Created Subscription: csi-addons-v0-8-0-sub  
INFO[0050] Approved InstallPlan install-mpsh4 for the Subscription: csi-addons-v0-8-0-sub 
INFO[0050] Waiting for ClusterServiceVersion "openshift-storage/csi-addons.v0.8.0" to reach 'Succeeded' phase 
INFO[0050]   Waiting for ClusterServiceVersion "openshift-storage/csi-addons.v0.8.0" to appear 
INFO[0053]   Found ClusterServiceVersion "openshift-storage/csi-addons.v0.8.0" phase: Pending 
INFO[0056]   Found ClusterServiceVersion "openshift-storage/csi-addons.v0.8.0" phase: Installing 
INFO[0066]   Found ClusterServiceVersion "openshift-storage/csi-addons.v0.8.0" phase: Succeeded 
INFO[0066] OLM has successfully installed "csi-addons.v0.8.0" 
INFO[0025] Creating a File-Based Catalog of the bundle "quay.io/ocs-dev/ocs-client-operator-bundle:release-4.16-7c5526d" 
INFO[0029] Generated a valid File-Based Catalog         
INFO[0032] Created registry pod: quay-io-ocs-dev-ocs-client-operator-bundle-release-4-16-7c5526d 
INFO[0033] Created CatalogSource: ocs-client-operator-catalog 
INFO[0033] Created Subscription: ocs-client-operator-v4-16-0-sub 
INFO[0044] Approved InstallPlan install-5hbkk for the Subscription: ocs-client-operator-v4-16-0-sub 
INFO[0044] Waiting for ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" to reach 'Succeeded' phase 
INFO[0045]   Waiting for ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" to appear 
INFO[0046]   Found ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" phase: Pending 
INFO[0057]   Found ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" phase: InstallReady 
INFO[0058]   Found ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" phase: Installing 
INFO[0112]   Found ClusterServiceVersion "openshift-storage/ocs-client-operator.v4.16.0" phase: Succeeded 
INFO[0113] OLM has successfully installed "ocs-client-operator.v4.16.0" 
INFO[0020] Creating a File-Based Catalog of the bundle "quay.io/noobaa/noobaa-operator-bundle:master-20231217" 
INFO[0024] Generated a valid File-Based Catalog         
INFO[0030] Created registry pod: quay-io-noobaa-noobaa-operator-bundle-master-20231217 
INFO[0030] Created CatalogSource: noobaa-operator-catalog 
INFO[0031] Created Subscription: noobaa-operator-v5-15-0-sub 
INFO[0044] Approved InstallPlan install-s29w2 for the Subscription: noobaa-operator-v5-15-0-sub 
INFO[0044] Waiting for ClusterServiceVersion "openshift-storage/noobaa-operator.v5.15.0" to reach 'Succeeded' phase 
INFO[0045]   Waiting for ClusterServiceVersion "openshift-storage/noobaa-operator.v5.15.0" to appear 
INFO[0046]   Found ClusterServiceVersion "openshift-storage/noobaa-operator.v5.15.0" phase: Pending 
INFO[0051]   Found ClusterServiceVersion "openshift-storage/noobaa-operator.v5.15.0" phase: Installing 
INFO[0057]   Found ClusterServiceVersion "openshift-storage/noobaa-operator.v5.15.0" phase: Succeeded 
INFO[0057] OLM has successfully installed "noobaa-operator.v5.15.0" 
INFO[0022] Creating a File-Based Catalog of the bundle "quay.io/mparida/ocs-operator-bundle:latest" 
INFO[0026] Generated a valid File-Based Catalog         
INFO[0031] Created registry pod: quay-io-mparida-ocs-operator-bundle-latest 
INFO[0031] Created CatalogSource: ocs-operator-catalog  
INFO[0032] Created Subscription: ocs-operator-v4-17-0-sub 
INFO[0046] Approved InstallPlan install-cczkp for the Subscription: ocs-operator-v4-17-0-sub 
INFO[0046] Waiting for ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" to reach 'Succeeded' phase 
INFO[0047]   Waiting for ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" to appear 
INFO[0051]   Found ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" phase: Pending 
INFO[0053]   Found ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" phase: InstallReady 
INFO[0054]   Found ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" phase: Installing 
INFO[0070]   Found ClusterServiceVersion "openshift-storage/ocs-operator.v4.17.0" phase: Succeeded 
INFO[0070] OLM has successfully installed "ocs-operator.v4.17.0" 
deployment.apps/rook-ceph-operator condition met
deployment.apps/csi-addons-controller-manager condition met
deployment.apps/ocs-client-operator-controller-manager condition met
deployment.apps/noobaa-operator condition met
deployment.apps/ocs-operator condition met
```
```
~ $ oc get csv
NAME                          DISPLAY                            VERSION   REPLACES   PHASE
csi-addons.v0.8.0             CSI Addons                         0.8.0                Succeeded
noobaa-operator.v5.15.0       NooBaa Operator                    5.15.0               Succeeded
ocs-client-operator.v4.16.0   OpenShift Data Foundation Client   4.16.0               Succeeded
ocs-operator.v4.17.0          OpenShift Container Storage        4.17.0               Succeeded
rook-ceph-operator.v4.17.0    Rook-Ceph                          4.17.0               Succeeded
```

```
~ $ oc get pods
NAME                                                              READY   STATUS      RESTARTS   AGE
44e38ffb37d6acbb78a40ec245de235eed70e40962720bf8719d63fc4adqpq8   0/1     Completed   0          6m13s
72fafb504211226f2e6a527f890af13cd36b2f3056ce6f1b8058dc4606glczh   0/1     Completed   0          5m13s
81a84b9d4356fe39b7ab535c6a493c0c8488e2b1defe37709a537297daps4lv   0/1     Completed   0          7m26s
8f27d12aa0e1f66007589b0e42d15222b5101365f865b538dafd7584e5jzk4w   0/1     Completed   0          2m21s
c96c99d0fcf6e276efaacf1088bb422b1a69f3e65e816bf8829019b2a5crvld   0/1     Completed   0          3m19s
csi-addons-controller-manager-69dfc94654-cqf67                    2/2     Running     0          5m59s
noobaa-operator-5795dcbd6d-hjjfm                                  2/2     Running     0          3m4s
ocs-client-operator-console-7778bfc485-xz99t                      1/1     Running     0          4m50s
ocs-client-operator-controller-manager-6465cbb94b-q2cgr           2/2     Running     0          4m50s
ocs-operator-7bcdd7b44b-284r7                                     1/1     Running     0          2m3s
quay-io-csiaddons-k8s-bundle-v0-8-0                               1/1     Running     0          6m16s
quay-io-mparida-ocs-operator-bundle-latest                        1/1     Running     0          2m27s
quay-io-noobaa-noobaa-operator-bundle-master-20231217             1/1     Running     0          3m28s
quay-io-ocs-dev-ocs-client-operator-bundle-release-4-16-7c5526d   1/1     Running     0          5m16s
rook-ceph-operator-d56ccf6f9-q6w2z                                1/1     Running     0          117s
ux-backend-server-69bc58577c-vxzzx                                2/2     Running     0          2m3s
```

File: Makefile [hunk 2]
```diff
@@ -180,6 +180,10 @@ else
 endif
 export KUSTOMIZE
 
+install: operator-sdk
```
Comment (malayparida2000) [inline Makefile:183]: May be we can call the target install-all ?

File: Makefile [hunk 3]
```diff
@@ -180,6 +180,10 @@ else
...
+	@echo "Installing operators"
+	hack/install-ocs-operator.sh
```
Comment (malayparida2000) [inline Makefile:185]: As we have already a hack file install-ocs.sh, how about we call it install-all/install-all-operators.sh ?

Comment (iamniting) [reply Makefile:185]: I will remove all those install scripts once this goes in. We will need to make a change in the release repo and after that we can remove that.

File: Makefile (hunk 2 above)
Comment (iamniting) [reply Makefile:183]: once we remove other targets in the coming PR we will have only one target which is called install.

**Review by malayparida2000 — APPROVED** [review]
/lgtm

**Pull request merged**